File: C/Loops2.c
#include <stdio.h>
void  Floyds_Triangle(int size){
    int bit=1;
    for (int row = 1; row <= size; row++)
    {
        bit=(row%2==0)?0:1;
        for (int column = 1; column <= row; column++)
        {
            printf("%d",bit);
            bit=(bit==0)?1:0;
        }
        printf("\n");
    }
}
void Pascals_Triangle(int size){
    for (int row = 1; row <= size; row++)
    {
        
        for (int column = 1; column <= row; column++)
        {
            printf("1");
            
        }
        printf("\n");
    }

}
void Pyramid_stars(int height){
    for (int i = 1; i <= height; i++)
    {
        for (int j = 1; j <= height-i; j++)
        {
            printf(" ");
        }

        for (int k = 1; k <=i ; k++)
        {
            printf("* ");
        }
        printf("\n");      
    }
    
}
void Pyramid_Increasing_Numbers(int height){
    int num=1;
    for (int i = 1; i <= height; i++)
    {
        for (int j = 1; j <= height-i; j++)
        {
            printf(" ");
        }

        for (int k = 1; k <=i ; k++)
        {
            printf("%d ",num++);
        }
        printf("\n");      
    }
    
}
int main() { 
    // Floyds_Triangle(5);
    // Pyramid_stars(4);
    Pyramid_Increasing_Numbers(4);
}

File: Java/Loops2.java
public class Loops2 {
    public static void main(String[] args) {
        // int []arr={1,9,2,4,5,3};
        // printStarsForArray(arr);
        // PrintStarGridSkipEven(10);
        // PrintStarsOnlyAtBorders(10);
        //PrintPyramid(4);
        PrintInvertedPyramid(4);
    }

    public static void printStarsForArray(int []arr){

        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr[i]; j++) {
                System.out.print("*");
            }
            System.out.println();
        }

    }

    public static void PrintStarGridSkipEven(int gridSize){
        for (int count=1; count <= gridSize; count++)
        {
            StringBuilder line = new StringBuilder();
            for (int row=1; row<= gridSize; row++)
            {
                // How to check if row value is even
                /* 
                if (row % 2 == 0)
                {
                    line.append(" ");
                }
                else
                {
                    line.append("*");
                } */
                line.append(row % 2 == 0 ? " " : "*");
            }
            System.out.println(line.toString());    
        }
    }

    public static void PrintStarsOnlyAtBorders(int gridSize){
        for (int row = 1; row <= gridSize; row++) {
            StringBuilder line=new StringBuilder();
             for (int column = 1; column <= gridSize; column++) {
                boolean isStar=row ==1 || row==gridSize || column==1 || column==gridSize;
                line.append(isStar?"* ":"  ");
            }
            System.out.println(line);
        }
    }

    public static void PrintPyramid(int height)
    {
        int countOfStars = 1;
        for (int level=1; level <= height; level++)
        {
            int noOfSpace = height-level;
            StringBuilder line = new StringBuilder();
            for (int count=0; count<noOfSpace; count++)
            {
                line.append(" ");
            }

            for (int count=0; count<countOfStars; count++)
            {
                line.append("*");
            }

            System.out.println(line.toString());
            countOfStars = countOfStars + 2;
        }
    }

     public static void PrintInvertedPyramid(int height)
    {
        int countOfStars = 2*height-1;
        for (int level=1; level <= height; level++)
        {
            int noOfSpace = level-1;
            // int noOfStars = ( height - level ) * 2 + 1; 
            StringBuilder line = new StringBuilder();
            for (int count=0; count<noOfSpace; count++)
            {
                line.append(" ");
            }

            for (int count=0; count<countOfStars; count++)
            {
                line.append("*");
            }

            System.out.println(line.toString());
            countOfStars = countOfStars - 2;
        }
    }
}
